pymesync.py

```diff
@@ -38,6 +44,42 @@ def __init__(self, baseurl, user, password, auth_type):
             "activity": [],
         }
 
+    def authenticate(self, username, password, auth_type):
+        """
+        authenticate(username, password, auth_type)
+
+        Authenticate a username and password with TimeSync via a POST request
+        to the login endpoint. This method will return a list containing a
+        single python dictionary. If successful, the dictionary will contain
+        the token in the form [{"token": "SOMETOKEN"}]. If an error is returned
+        the dictionary will contain the error information.
+
+        ``username`` is a string containing the username of the TimeSync user
+        ``password`` is a string containing the user's password
+        ``auth_type`` is a string containing the authentication method used by
+        TimeSync
+        """
+        self.user = username
+        self.password = password
+        self.auth_type = auth_type
+        auth = {"auth": self._auth()}
+        url = "{}/login".format(self.baseurl)
+
+        try:
+            # Success!
+            response = requests.post(url, json=auth)
+            token_list_dict = self._json_to_python(response.text)
+        except requests.exceptions.RequestException as e:
+            # Request error
+            return [{self.error: e}]
+
+        if "error" in token_list_dict[0]:
+            return token_list_dict
+        else:
+            token_dict = token_list_dict[0]
+            self.token = token_dict["token"]
```

Comment:
You can, if you want, just string these together:
self.token = token_list_dict[0]["token"]

Saves a variable, and I think the impact on readability is minimal.